%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ELIFE ARTICLE TEMPLATE
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% PREAMBLE 
\documentclass[9pt,lineno]{elife}
\sloppy
% Use the onehalfspacing option for 1.5 line spacing
% Use the doublespacing option for 2.0 line spacing
% Please note that these options may affect formatting.

\usepackage{lipsum} % Required to insert dummy text
\usepackage[version=4]{mhchem}
\usepackage{siunitx}
\DeclareSIUnit\Molar{M}
%\usepackage{indentfirst} %I prefer to have indents following section headings, I find it to be more consistent in style throughout document
\usepackage{natbib}
\usepackage{float}
\usepackage{tikz}


\newfloat{suppfile}{thp}{lofsupfile}
%\renewcommand{\thesuppfile}{Supplementary file \arabic{suppfile}}
\floatname{suppfile}{Supplementary file}

\renewcommand{\floatpagefraction}{1}

\newcommand{\jdbcomment}[1]{\emph{\color{red} [#1]}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE SETUP
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Extreme heterogeneity of influenza virus infection in single cells}

\author[1]{Alistair B. Russell}
\author[2]{Cole Trapnell}
\author[1,2*]{Jesse D. Bloom}
\affil[1]{Basic Sciences Division and Computational Biology Program, Fred Hutchinson Cancer Research Center, Seattle, United States}
\affil[2]{Department of Genome Sciences, University of Washington, Seattle, United States}
\corr{jbloom@fredhutch.org}{}

% \presentadd[\authfn{5}]{eLife Sciences editorial Office, eLife Sciences, Cambridge, United Kingdom}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%% ARTICLE START
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

\maketitle

\begin{abstract}
Viral infection can dramatically alter a cell's transcriptome.
However, these changes have mostly been studied by bulk measurements on many cells.
Here we use single-cell mRNA sequencing to examine the transcriptional consequences of influenza virus infection.
We find extremely wide cell-to-cell variation in the productivity of viral transcription -- viral transcripts comprise less than a percent of total mRNA in many infected cells, but a few cells derive over half their mRNA from virus.
Some infected cells fail to express at least one viral gene, but this gene absence only partially explains variation in viral transcriptional load.
Despite variation in viral load, the relative abundances of viral mRNAs are fairly consistent across infected cells.
Activation of innate immune pathways is rare, but some cellular genes co-vary in abundance with the amount of viral mRNA.
Overall, our results highlight the complexity of viral infection at the level of single cells.
\end{abstract}

\section{Introduction}
Viruses can cause massive and rapid changes in a cell's transcriptome as they churn out viral mRNAs and hijack cellular machinery.
For instance, cells infected with influenza virus at high multiplicity of infection (MOI) express an \emph{average} of 50,000 to 100,000 viral mRNAs per cell, corresponding to 5 to 25\% of all cellular mRNA~\citep{hatada1989control}.
Infection can also trigger innate-immune sensors that induce the expression of cellular anti-viral genes~\citep{Killip:2015dw,Iwasaki:2014dw,crotta2013type}.
This anti-viral response is another prominent transcriptional signature of high-MOI influenza virus infection in bulk cells~\citep{Geiss:2002a}.

However, initiation of an actual influenza infection typically involves just a few virions infecting a few cells~\citep{varble2014influenza,poon2016quantifying,leonard2017transmission,mccrone2017evolutionary}. 
The dynamics of viral infection in these individual cells may not mirror bulk measurements made on many cells infected at high MOI.
Over 70 years ago, Max Delbruck showed that there was a $\sim$100-fold range in the number of progeny virions produced per cell by clonal bacteria infected with clonal bacteriophage~\citep{Delbruck:1945wo}.
Subsequent work has shown similar heterogeneity during infection with other viruses~\citep{zhu2009growth,Schulte:2014br,combe2015single,Akpinar:2016gt}, including influenza virus~\citep{Heldt:2015iz}.

In the case of influenza virus infection, targeted measurements of specific proteins or RNAs have shed light on some factors that contribute to cell-to-cell heterogeneity. 
The influenza virus genome consists of eight negative-sense RNA segments, and many infected cells fail to express one more of these RNAs~\citep{Heldt:2015iz,Dou:2017cp} or their encoded proteins~\citep{Brooke:2013kb}.
In addition, activation of innate-immune responses is inherently stochastic~\citep{shalek2013single,shalek2014single,bhushal2017cell,hagai2017gene}, and only some influenza-infected cells express anti-viral interferon genes~\citep{PerezCidoncha:2014jr,Killip:2017ef}.
However, the extent of cell-to-cell variation in these and other host and viral factors remains unclear, as does the association among them in individual infected cells.

Here we use single-cell mRNA sequencing to quantify the levels of all cellular and viral mRNAs in cells infected with influenza virus at low MOI.
We find extremely large variation in the amount of viral mRNA expressed in individual cells.
Both co-infection and activation of innate-immune pathways are rare in our low-MOI infections, and do not appear to be the major drivers of cell-to-cell heterogeneity in viral transcriptional load.
Individual infected cells often fail to express specific viral genes, and such gene absence explains some but certainly not all of the cell-to-cell heterogeneity.
A variety of cellular genes, including ones involved in the oxidative-stress response, co-vary with viral transcriptional load.
Overall, our work demonstrates remarkable heterogeneity in the transcriptional outcome of influenza virus infection among nominally identical cells infected with a relatively pure population of virions.

\section{Results}
\begin{figure}
\centerline{\includegraphics[width=0.8\linewidth]{figures/Workflow/workflow.pdf}}
\caption{\label{fig:workflow} Experimental design.
{\bf (A)}  
We engineered a virus that carried two synonymous mutations near the 3' end of each mRNA.
At top are the mutations for PB2.
At bottom are locations of the synonymous mutations relative to the typical distribution of read depth for our 3'-end sequencing.
{\bf (B)} 
The wild-type and synonymously barcoded viruses transcribe their genes with similar kinetics. 
The abundance of the viral hemagglutinin (HA) transcript relative to the cellular housekeeping gene L32 was assessed by qPCR in A549 cells infected at an MOI of 0.5 (as determined on MDCK-SIAT1 cells).
Error bars $\pm$ S.D., n=3.
{\bf (C)}  
For the single-cell mRNA sequencing, A549 cells were infected with an equal mixture of wild-type and synonymously barcoded virus. 
Immediately prior to collection, cells were physically separated into droplets and cDNA libraries were generated containing the indicated barcodes. 
The libraries were deep sequenced, and the data processed to create a matrix that gives the number of molecules of each transcript observed in each cell.
Infected cells were further annotated by whether their viral mRNAs derived from wild-type virus, synonymously barcoded virus, or both.
}
\figdata{Sequences of wild-type and barcoded viruses are in \texttt{viralsequences.fasta}.\label{figdata:viralsequences}}
\end{figure}

\subsection{Strategy to measure mRNA in single virus-infected cells.}
We performed single-cell mRNA sequencing using a droplet-based system that physically isolates individual cells prior to reverse transcription~\citep{zheng2017massively,macosko2015highly,Klein:2015ki}.
Each droplet contains primers with a unique \emph{cell barcode} that tags all mRNAs from that droplet during reverse-transcription.
Each primer also contains a \emph{unique molecular identifier (UMI)} that is appended to each mRNA molecule during reverse transcription.
The 3' ends of the mRNAs are sequenced and mapped to the human and influenza virus transcriptomes to determine transcript identities.
This information is combined with that provided by the UMIs and cell barcodes to quantify the number of molecules of each mRNA species that have been captured for each cell.

Infected cells will express viral as well as cellular mRNAs -- however the cell barcodes and UMIs cannot distinguish whether a cell was initially infected by one or multiple viral particles.
We therefore engineered an influenza virus (strain A/WSN/1933) that additionally carried \emph{viral barcodes} consisting of synonymous mutations near the 3' end of each transcript (\FIG{workflow}A).
Critically, these synonymous mutations did not greatly impact viral growth kinetics (\FIG{workflow}B).
We infected A549 human lung carcinoma cells with an equal mix of the wild-type and synonymously barcoded viruses.
Cells infected by a single virion will exclusively express mRNAs from either wild-type or synonymously barcoded virus, whereas cells that are co-infected with multiple virions will often express mRNAs from both the wild-type and synonymously barcoded viruses (\FIG{workflow}C).

We took care to generate stocks of virus that were relatively ``pure'' of defective particles.
Stocks of viruses typically contain an array of biologically active viral particles, some of which are defective for replication owing to mutations or deletions in essential viral genes~\citep{von1954incomplete,huang1970defective,Brooke:2014ch,Fonville:2015cg,Lauring:2010if,Dimmock:2014dk,saira2013sequence}.
These defective particles become prevalent when a virus is grown at high MOI, where complementation permits the growth of otherwise deleterious genotypes.
To minimize the levels of defective particles, we propagated our viral stocks at low MOI for a relatively brief period of time~\citep{Xue:2016bw}.
We validated that our stocks exhibited greater purity of infectious particles than a stock propagated at high MOI by verifying that they had a higher ratio of infectious particles to virion RNA (\FIG{viruspopulations}A) and to particles capable of inducing expression of a single viral protein (\FIG{viruspopulations}B).
In addition, viral stocks with many defective particles are more immunostimulatory per infectious unit (e.g., TCID50) than low-defective stocks~\citep{Tapia:2013kf,Lopez:2014en}, in part simply because there are more physical virions per infectious unit (\FIG{viruspopulations}A,B).
We confirmed that our viral stocks induced less interferon per infectious unit than a stock propagated at higher MOI (\FIG{viruspopulations}C).

\begin{figure}
\centerline{\includegraphics[width=0.7\linewidth]{figures/Validating_barcode_virus/validating_populations_D02.pdf}}
\caption{\label{fig:viruspopulations} The viral stocks in our experiments are relatively pure of defective particles. 
{\bf (A)}
Our viral stocks have a higher ratio of infectious particles to HA virion RNA compared to a high-defective stock propagated at high MOI.
HA viral RNA was quantified by qPCR on virions. 
Error bars $\pm$ S.D., n=6 (qPCR replicates). 
{\bf (B)} 
Our viral stocks have a higher ratio of infectious particles to particles capable of expressing HA protein.
A549 cells were infected at an MOI of 0.1, and the percentage of cells expressing HA protein at 9 hours post-infection was quantified by antibody staining and flow cytometry.
{\bf (C)} 
Our viral stocks are less immunostimulatory than virus propagated at high MOI when used at the same number of infectious units as calculated by TCID50.
Note that this fact does not necessarily imply that they are more immunostimulatory per virion, as the high-MOI stocks also have more virions per infectious unit as shown in the first two panels. 
Measurements of \textit{IFNB1} transcript by qPCR normalized to the housekeeping gene L32 in A549 cells at 10 hours post infection at an MOI of 0.5.
Error bars $\pm$ S.D., n=3.
Note that MOIs were calculated by TCID50 on MDCK-SIAT1 cells, whereas the experiments in this figure involved infection of A549 cells.
}
\figsupp[Full flow cytometry data for panel B . \label{figsupp:HAstain}]{
Full flow cytometry data for  \FIG{viruspopulations}B. 
A549 cells were infected at an MOI of 0.1 as calculated by TCID50 on MDCK-SIAT1 cells. 
{\bf (A)} Uninfected gating control.
{\bf (B)} Cells infected with the wild-type virus stock used in our experiments.
{\bf (C)} Cells infected with synonymously barcoded virus stock used in our experiments.
{\bf (D)} Cells infected with a stock of wild-type virus propagated at a high MOI, and therefore enriched in defective particles.
}{\includegraphics[width=0.7\linewidth]{figures/Validating_barcode_virus/HA_stain_supplement.png}}
\end{figure}

\subsection{Single cells show an extremely wide range of expression of viral mRNA.}
We infected A549 cells at low MOI with a mixture of the wild-type and synonymously barcoded viruses, and collected cells for sequencing at 6, 8, and 10 hours post-infection, performing two slightly different variants of the experiment for the 8-hour timepoint.
For most of the samples, we replaced the infection inoculum with fresh media at one-hour post-infection, thereby ensuring that most infection was initiated during a narrow time window.
However, for the second 8-hour sample (which we denote as ``8hr-2'' in the figures), we did \emph{not} perform this media change and instead left the cells in the original infection inoculum.
The rationale for including a sample without a media change was to determine the importance of synchronicity of the timing of infection as discussed later in this subsection.

We recovered between 3,000 and 4,000 cells for each sample (\FIG{cells}A). 
As expected for a low-MOI infection, most cells expressed little or no viral mRNA (\FIG{cells}B, \FIGSUPP[cells]{viral_mRNA_cumulfrac}).
Also as expected, the amount of viral mRNA per cell among infected cells increased over time (\FIG{cells}B, \FIGSUPP[cells]{viral_mRNA_cumulfrac}).
But what was most notable was how widely the number of viral mRNA molecules varied among infected cells.
While the fraction of mRNA derived from virus was $<$0.1\% for most cells, viral mRNA constituted half the transcriptome in a few cells at 8 and 10 hours (\FIG{cells}B, \FIGSUPP[cells]{viral_mRNA_cumulfrac}).

\begin{figure}
\centerline{\includegraphics[width=\linewidth]{figures/p_cell_mRNA_summary.pdf}}
\caption{\label{fig:cells}
There is a very wide distribution in the amount of viral mRNA per cell.
{\bf (A)} 
Number of cells sequenced for each sample.
{\bf (B)} 
The number of cellular and viral mRNAs detected for each cell is plotted as a point.
The blue lines show the overall distribution of the number of cellular mRNAs per cell.
The orange rug plot at the left of each panel shows the distribution of the number of viral mRNAs per cell.
Cells outside the dotted green lines were considered outliers with suspiciously low or high amounts of cellular mRNA (possibly derived from two cells per droplet), and were excluded from all subsequent analyses.
\FIGSUPP[cells]{viral_mRNA_cumulfrac} shows the exact distributions of the fraction of viral mRNA per cell.
}
\figsupp[Cumulative fraction plot of proportion of total mRNA from virus. \label{figsupp:viral_mRNA_cumulfrac}]{
For each sample, this plot shows the fraction of all cells that derive at least the indication fraction of their mRNA from influenza virus.
}{\includegraphics[width=0.5\linewidth]{figures/p_mRNA_per_cell_cumul.pdf}}
\end{figure}

A complicating factor is that uninfected cells could have small amounts of viral mRNA due to leakage of transcripts from lysed cells.
It is therefore important to establish a threshold for identifying truly infected cells.
We can do this by taking advantage of the fact that roughly half the infecting virions bear synonymous barcodes.
Reads derived from lysed cells will be drawn from both wild-type and synonymously barcoded viral transcripts.
However, most cells are infected by at most one virion, and so the reads from truly infected cells will usually derive almost entirely from one of the two viral variants. 
\FIG{viralbarcodes}A shows the fraction of viral reads in individual cells from each viral variant, and \FIG{viralbarcodes}B indicates the fraction of viral reads from the most abundant variant in that cell.
Most cells with large amounts of viral mRNA have viral transcripts exclusively derived from one viral variant -- indicating non-random partitioning as expected from viral infection.
However, cells with a small amount of viral mRNA often have viral transcripts from both variants, as expected from the random partitioning associated with simple mRNA leakage.
Finally, a few cells with large amounts of viral mRNA have viral transcripts from both variants, likely reflecting co-infection.

\begin{figure}[t!]
\centerline{\includegraphics[width=0.82\linewidth]{figures/p_frac_flu_summary.pdf}}
\caption{\label{fig:viralbarcodes}
Synonymous barcodes on the viral mRNAs distinguish true infections from cells that contain viral mRNAs derived from leakage of lysed cells.
{\bf (A)}
Cells with at least two viral mRNAs for which the barcode could be called, arranged in order of increasing influenza transcript counts.
Bar heights denote the number viral mRNAs on a log\textsubscript{10} scale, bar coloring is linearly proportional to the fractions of viral mRNAs derived from wild-type and synonymously barcoded virus.
{\bf (B)}
Same as (A), but each bar is colored according to the relative fraction of the more common (major) and less common (minor) virus variant.
At low levels of viral mRNA there is often a roughly equal mix, suggesting contamination with viral mRNAs leaked from lysed cells.
At higher levels of viral mRNA, cells generally have only one viral variant, suggesting infection initiated by a single virion.
A few cells are also obviously co-infected with both viral variants.
{\bf (C)}
We determined a threshold for calling ``true'' infections by finding the amount of viral mRNA per cell at which the viral barcode purity no longer increases with more viral mRNA.
The purity is the fraction of all viral mRNA in a cell derived from the most abundant viral barcode in that cell.
We fit a curve (orange line) to the mean purity of all cells with more than the indicated amount of viral mRNA, and drew the cutoff (dotted green line) at the point where this curve stopped increasing with the fraction of total mRNA derived from virus.
This plot illustrates the process for the 10-hour sample, see \FIGSUPP[viralbarcodes]{thresholds} for similar plots for other samples.
See the Methods for details.
{\bf (D)}
The number of cells identified as infected and co-infected for each sample, as well as the number of cells with any viral read.
For all subsequent analyses, we subsampled the number of uninfected cells per sample to the greater of 50 or the number of infected cells.
{\bf (E)} 
Distribution of the fraction of mRNA per cell derived from virus for both infected and co-infected cells.
\FIGSUPP[viralbarcodes]{cumulfracflu} shows these same data in a cumulative fraction plots and calculates Gini coefficients to quantify the heterogeneity in viral mRNA load.
}

\figsupp[Number of viral barcodes called.\label{figsupp:barcodescalled}]{
The number of viral barcodes called for each sample and gene segment. 
Viral transcripts are classified as \emph{syn} if they mapped to a synonymously barcoded influenza transcript, \emph{wt} if they mapped to a wild-type influenza transcript, \emph{invalid} if multiple reads for the same UMI differed on the status of the viral barcode, and as \emph{uncalled} if none of the reads for that UMI overlapped the region of the viral transcript containing the viral barcode.
For calculations of the number of reads in a cell derived from each viral barcode for each viral gene, the total number of detected molecules of that gene are multiplied by the fraction of those molecules with assignable barcodes that are assigned to that barcode.
}{\includegraphics[width=\linewidth]{figures/synbarcodes_umistats.jpg}}

\figsupp[Thresholds for calling infected cells. \label{figsupp:thresholds}]{
Cell lysis can lead cells to the spurious association of small amounts of extraneous mRNA with individual cells.
We wanted to avoid classifying as infected cells that had simply acquired such lysis-derived viral mRNA.
The amount of lysis-derived viral mRNA will vary among samples as a function of both the lysis rate during the cell preparation (which always varies slightly from sample to sample in the 10X procedure) and with the amount of total viral mRNA for that sample (the more viral mRNA, the more there is to be acquired from lysed cells).
As is shown in \FIG{viralbarcodes}B, the 8hr-2 and 10hr sample clearly have an enrichment of mixed barcodes in cells with small numbers of viral mRNA.
For each sample, we calculated the mean purity of all cells with at least the indicated amount of viral mRNA, and determined the threshold amount of viral mRNA where purity no longer increased by finding the first maxima in a loess curve fit (orange line).
We called the threshold at this point of maximum purity (dotted green line).
For the 6hr and 8hr samples there is no indication of contamination from lysis-derived reads, as \FIG{viralbarcodes}B shows no increase in mixed barcodes in low viral mRNA cells.
Therefore, for these samples we simply set a threshold of requiring at least 2$\times$10$^{-4}$ of the total mRNA to come from virus, which corresponds to $\sim$2 viral mRNAs for the typical cell with $10^4$ total reads (\FIG{cells}B).
}{\includegraphics[width=\linewidth]{figures/p_purity_threshold.pdf}}

\figsupp[Cumulative distributions of viral mRNA per cell and Gini coefficients.\label{figsupp:cumulfracflu}]{
The total fraction of all viral mRNA among infected cells that is attributable to a given fraction of these cells.
For instance, the plot for the 8hrs sample shows that $\sim$50\% of all viral mRNA is derived from $\sim$8\% of the infected cells.
The facet titles above each plot also give the Gini coefficient~\citep{gini1921measurement} that calculates the heterogeneity in the distribution of viral mRNA among infected cells.
Gini coefficients of 0 indicate a perfectly even distribution across cells, and Gini coefficients of 1 indicate a maximally skewed distribution.
}{\includegraphics[width=\linewidth]{figures/p_cumul_flu.pdf}}

\figsupp[Synchronization of infection does not greatly affect heterogeneity. \label{figsupp:synchronize}]
{Flow cytometry analysis of expression of viral proteins in cells infected at an MOI of 0.1 (unsynchronized) or 0.2 (synchronized) as calculated by TCID50 on MDCK-SIAT1 cells. 
The higher MOI for synchronized infections was to attempt to account for loss of virus in washing steps.
Infections were synchronized by pre-adsorbing virus at 4$^{\circ}$C for 1h prior to initiation of infection by shifting temperature to 37$^{\circ}$C using pre-warmed media.
Cells were concurrently stained for HA and NS1 proteins at 10 hours after initiation of infection, and then analyzed by flow cytometry.
The level of HA protein was quantified in cells that were identified as infected based on being positive for NS1 protein (top), and the level of NS1 protein was quantified in cells that were identified as infected based on being positive for HA protein (bottom)
Synchronization resulted in a small decrease in variability in the levels of each viral protein, particularly in cells with intermediate levels.
But the effects were small compared to the overall variability in the protein levels, indicating timing of infection makes only a small contribution to the observed heterogeneity. 
Data are shown for three independent replicates.
}{\includegraphics[width=\linewidth]{figures/synchronize.jpg}}

\figsupp[Effects of infectious dose or coinfection state. \label{figsupp:flowdose}]
{
{\bf (A)}
Flow cytometry analysis of expression of viral proteins in cells infected at high (MOI 5 as calculated by TCID50 on MDCK-SIAT1 cells) or low (MOI 0.1) initial infectious dose.
Cells were concurrently stained for HA and NS1 proteins at 10 hours after initiation of infection, and then analyzed by flow cytometry.
The level of HA protein was quantified in cells that were identified as infected based on being positive for NS1 protein (left), and the level of NS1 protein was quantified in cells that were identified as infected based on being positive for HA protein (right)
While a higher dose leads to more cells expressing high amounts of viral protein, it does not greatly increase the amount of viral protein in either the low-expressing or high-expressing cells. 
Therefore, higher viral dose does not lead to a large continuous increase in viral protein production among all cells -- rather, it mostly changes the proportions of cells that fall in different parts of the highly heterogeneous distribution.
{\bf (B)}
Cells were co-infected with a mix of wild-type virus and pseudovirus in which the HA gene was replaced by GFP flanked by the terminal regions of the HA gene segment at an MOI of 0.1 for each virus. 
At 10 hours post-infection, cells were stained for NS1 and HA expression and analyzed by flow cytometry for these proteins and GFP.
Cells could be annotated as infected by virions of the same type (wild-type infection indicated by presence of HA, or pseudovirus infection indicated by the presence of GFP) or both types of virions (indicated by presence of HA and GFP).
Coinfected cells, like cells infected at a higher infectious dose, occupy different positions in the distribution of viral protein production but do not not exhibit a continuous increase in viral protein production.
Data are shown for three independent replicates.
}{\includegraphics[width=\linewidth]{figures/Coinf.jpg}}
\end{figure}

We determined the threshold amount of viral mRNA per cell for each sample at which the barcode partitioning clearly resulted from infection rather than leakage (\FIG{viralbarcodes}C, \FIGSUPP[viralbarcodes]{thresholds}), and used these thresholds to annotate cells that we were confident were truly infected.
We also annotated as co-infected cells above this threshold that had mRNA from both viral variants.
\FIG{viralbarcodes}D shows the number of cells annotated as infected and co-infected for each sample -- these cells are just a small fraction of the number of cells with any viral read.
These annotation thresholds are conservative, and may miss some true low-level infections.
However, it is important that the analyses below are restricted to cells that are truly infected with virus, so we accepted the possible loss of some low-level infections in order to avoid false positives.
In addition, the synonymous viral barcodes only identify co-infections by viruses with different barcodes -- since the barcodes are at roughly equal proportion, we expect to miss about half of the co-infections.
Since we annotate about $\sim$10\% of the infected cells as co-infected by viruses with different barcodes (\FIG{viralbarcodes}D), we expect another $\sim$10\% of the infected cells to also be co-infected but not annotated as so by our approach.
Because most cells are not infected, we subsampled the uninfected cells to the numbers shown in \FIG{viralbarcodes}D to balance the proportions of infected and uninfected cells for all subsequent analyses.

Strikingly, the extreme variation in the number of viral transcripts per cell remains even after we apply these rigorous criteria for annotating infected cells (\FIG{viralbarcodes}E). 
The fraction of viral mRNA per infected cell follows a roughly exponential distribution, with many cells having few viral transcripts and a few cells having many.
At 6 and 8 hours $<$10\% of infected cells are responsible for over half the viral transcripts, while at 10 hours $<$15\% of infected cells produce over half the viral transcripts (\FIGSUPP[viralbarcodes]{cumulfracflu}).
One way to quantify the heterogeneity of a distribution is to calculate the Gini coefficient~\citep{gini1921measurement}, which ranges from 0 for a completely uniform distribution, to 1 for a maximally skewed distribution.
\FIGSUPP[viralbarcodes]{cumulfracflu} shows the Gini coefficients for the distribution of viral mRNA across infected cells for each sample.
The Gini coefficients are $\ge$0.64 for all samples.
As a fun point of comparison, these Gini coefficients indicate that the distribution of viral mRNA across infected cells is more uneven than the distribution of income in the United States~\citep{alvaredo2011note}. 

One possible source of heterogeneity in the amount of viral mRNA per cell is variability in the timing of infection.
If some cells are infected earlier in the experiment than others, then they might have substantially more viral mRNA.
However, several lines of evidence indicate that this is not the major cause of heterogeneity across cells.
First, the sample for which the infection inoculum was never removed (8hr-2) only shows slightly more heterogeneity than samples for which the inoculum was washed away after one hour (\FIG{viralbarcodes}E, \FIGSUPP[viralbarcodes]{cumulfracflu}), despite the fact that the potential time window for infection is much longer in the former sample.
Second, in an independent experiment, we performed completely synchronized infections by pre-binding virus to cells on ice and then washing away unbound virus before bringing the cells to 37$^{\circ}$C~\citep{Dapat:2014kv}.
As shown in \FIGSUPP[viralbarcodes]{synchronize}, flow cytometry staining found that the heterogeneity in the levels of individual viral proteins was not markedly different for these synchronized infections than in the absence of pre-binding and washing.
Finally, viral mRNA expression from the secondary spread of virus from infected cells does not appreciably occur during the timeframes of our experiments, since \FIG{viralbarcodes}B does not show the pervasive presence of mixed barcodes that would occur in this case.
Therefore, variability in the timing of infection is not the dominant cause of the cell-to-cell heterogeneity in our experiments.

Notably, \FIG{viralbarcodes}E shows that there are co-infected cells with both low and high amounts of viral mRNA, suggesting that the initial infectious dose does not drive a simple continuous increase in viral transcript production.
In support of this view, we used flow cytometry to quantify the levels of individual viral proteins in cells infected at various MOIs or for which we could delineate co-infection status (\FIGSUPP[viralbarcodes]{flowdose}).
This analysis shows that sub-populations of cells that express similarly low and high levels of viral proteins persist across a wide range of infectious doses, although co-infection can influence the relative proportion of infected cells that fall into these sub-populations (\FIGSUPP[viralbarcodes]{flowdose}).

\subsection{Absence of viral genes partially explains cell-to-cell variability in viral load.}
The influenza genome is segmented, and cells can fail to express a viral mRNA if the encoding gene segment is not packaged in the infecting virion or fails to initiate transcription after infection.
Indeed, several groups have reported that the majority of infected cells fail to express at least one viral gene~\citep{Brooke:2013kb, Heldt:2015iz,Dou:2017cp}. 
We wondered if the absence of specific viral genes might be associated with reduced amounts of viral mRNA within single infected cells.
In particular, transcription of influenza virus mRNAs is performed by the viral ribonucleoprotein (RNP) complex, which consists of the three proteins that encode the tripartite polymerase (PB2, PB1, and PA) as well as nucleoprotein (NP)~\citep{huang1990determination}.
Each viral gene segment is associated with one RNP in incoming infecting virions, but secondary transcription by newly synthesized RNPs requires the presence of the viral genes encoding each of the four RNP proteins~\citep{Vreede:2004ip,eisfeld2015centre}.
This secondary transcription is a major source of viral mRNAs, as evidenced by the fact that blocking synthesis of the RNP proteins reduces the amount of viral mRNA by several orders of magnitude in bulk cells (\FIGSUPP[fluburdenbyflugene]{cyclohexamide}).

\begin{figure}[t!]
\centerline{\includegraphics[width=0.83\linewidth]{figures/p_flu_burden_flu_gene_merge.pdf}}
\caption{\label{fig:fluburdenbyflugene}
The absence of viral genes explains some of the variability in the amount of viral mRNA per cell.
{\bf (A)} 
The normalized expression of each viral gene as a function of the total fraction of mRNA in each infected cell derived from virus, taken over all time points.
Cells with high viral burden always express all RNP genes, but some cells with high viral burden lack each of the other genes.
Plots for individual samples are in \FIGSUPP[fluburdenbyflugene]{fluburdensamplebreakdown}, and a plot that excludes known coinfected cells is in \FIGSUPP[fluburdenbyflugene]{fluburden_nocoinfection}.
{\bf (B)}
Box and whisker plots showing the per-cell viral burden among cells with $>$0.5\% of their mRNA from virus, binned by whether or not the cells express each gene.
A Wilcoxon signed-rank test was used to test the null hypothesis that absence of each gene does not affect viral burden: **** = $P < 10^{-4}$, *** = $P < 10^{-3}$,  * = $P < 0.05$, ns = not significant.
The trends are similar if we look only at the 10-hour sample (\FIGSUPP[fluburdenbyflugene]{10hrfluburdenbyflugene}) or exclude known co-infected cells (\FIGSUPP[fluburdenbyflugene]{fluburdenbyflugene_nocoinfection}).
{\bf (C)}
The fraction of cells that express each of the four other genes among cells that express all RNP genes, as well as the fraction that express \emph{all} four of the other genes.
The fraction that express all four genes is well predicted by simply multiplying the frequencies of cells that express each gene individually, indicating that gene absence is approximately independent across these genes.
}

\figsupp[Secondary transcription is a major source of viral mRNA during bulk infections.\label{figsupp:cyclohexamide}]{
A549 cells were infected at an MOI of 0.2 as calculated on MDCK-SIAT1 cells in either the presence or absence of the protein-translation inhibitor cyclohexamide, and viral mRNA was quantified at 8 hours post-infection by qPCR.
The cyclohexamide prevents translation of new PB2, PB1, PA, and NP protein, and so prevents the formation of the new RNPs needed for secondary transcription.
The bars show the relative amount of HA and PB2 mRNA in the absence versus the presence of cyclohexamide. Error $\pm$ S.D. n=3. 
}{\includegraphics[width=0.3\linewidth]{figures/Primary_transcription_measurements/Cyclohexamide.pdf}}

\figsupp[Like panel (A), but shows samples individually.\label{figsupp:fluburdensamplebreakdown}]{
The normalized expression of each viral gene as a function of the fraction of total mRNA derived from virus, shown for the 10-hour and 8-hour samples individually (the other samples had too few infected cells for this analysis to be useful).
Points are colored as in \FIG{fluburdenbyflugene}A.
}{\includegraphics[width=0.8\linewidth]{figures/p_flu_burden_flu_gene_by_sample.pdf}}

\figsupp[Like panel (A), but excludes coinfected cells with mixed viral barcodes.\label{figsupp:fluburden_nocoinfection}]{
The normalized expression of each viral gene as a function of the fraction of total mRNA derived from virus, excluding cells that were annotated as coinfected based on the presence of both viral barcodes.
}{\includegraphics[width=0.8\linewidth]{figures/p_flu_burden_flu_gene_no_coinfect.pdf}}

\figsupp[Like panel (B) but for the 10-hr sample only.\label{figsupp:10hrfluburdenbyflugene}]{
The absence of viral RNP genes but \emph{not} non-RNP genes remains significantly associated with reduced viral burden when we examine only the 10-hr sample, which is the single time point with the most data points.
The difference for NP is no longer statistically significant due to low counts of infected cells lacking NP, but the trend remains.
We do not show statistical analyses for other samples, as the number of infected cells is too low.
}{\includegraphics[width=\linewidth]{figures/p_10hr_flu_burden_flu_gene_test}}

\figsupp[Like panel (B) but excludes coinfected cells with mixed viral barcodes.\label{figsupp:fluburdenbyflugene_nocoinfection}]{
All findings in \FIG{fluburdenbyflugene}B remain unchanged if we exclude cells called as coinfected based on the presence of mixed viral barcodes.
}{\includegraphics[width=\linewidth]{figures/p_no_coinfect_flu_burden_flu_gene_test}}

\figdata{\label{figdata:fracmissing} The numerical data for panel (C) are in \texttt{p\_missing\_genes.csv}.}

\figdata{\label{figdata:simplemodel} Simulation with a simple model for the expected heterogeneity due to Poisson co-infection and presence / absence of the full RNP is in \texttt{simple\_Poisson\_model.html}.}

\end{figure}

We examined the total amount of viral mRNA versus the expression of the genes from each viral segment (\FIG{fluburdenbyflugene}A, \FIGSUPP[fluburdenbyflugene]{fluburdensamplebreakdown}, \FIGSUPP[fluburdenbyflugene]{fluburden_nocoinfection}). 
Note that influenza virus expresses ten major gene transcripts from its eight gene segments, as the M and NS segments are alternatively spliced to produce the M1 / M2 and NS1 / NEP transcript, respectively~\citep{dubois2014influenza}.
However, an inherent limitation of current established single-cell mRNA sequencing techniques is that they only sequence the 3' end of the transcript~\citep{zheng2017massively,macosko2015highly,Klein:2015ki,cao2017comprehensive}.
Since the alternative spliceoforms M1 / M2 and NS1 / NEP share the same 3' ends, we cannot distinguish them and therefore will refer simply to the combined counts of transcripts from each of these alternatively spliced segments as the M and NS genes.

Cells that lack an RNP gene never derive more than a few percent of their mRNAs from virus, confirming the expected result that all four RNP genes are essential for high levels of viral transcription (\FIG{fluburdenbyflugene}A, \FIGSUPP[fluburdenbyflugene]{fluburdensamplebreakdown}, \FIGSUPP[fluburdenbyflugene]{fluburden_nocoinfection}).
However, we observe cells that lack each of the other non-RNP genes but still derive $\approx$40\% of their mRNAs from virus, suggesting that none of the other genes are important for high levels of viral transcription.
These results are statistically supported by \FIG{fluburdenbyflugene}B, which shows that absence of any RNP gene but \emph{not} any other viral gene is associated with reduced amounts of viral mRNA. 
However, gene absence clearly does not explain all of the variability in viral gene expression, since even cells expressing all viral genes exhibit a very wide distribution in the amount of viral mRNA that they express.
Specifically, at both 8 and 10 hours, the amount of viral mRNA in individual cells expressing all eight viral genes still ranges from $<$1\% to $>$50\% (\FIG{fluburdenbyflugene}A, \FIGSUPP[fluburdenbyflugene]{fluburdensamplebreakdown}, \FIGSUPP[fluburdenbyflugene]{fluburden_nocoinfection}).  
Furthermore, the actual distribution of viral mRNA per infected cell (\FIG{viralbarcodes}E) does not match the mostly bi-modal shape expected under a simple model where RNP gene absence and Poisson co-infection are the only factors (\FIG{fluburdenbyflugene}--source~data~\ref{figdata:simplemodel}), indicating that there are additional sources of variability beyond whether cells have full complement of RNP genes. 

We also quantified the fraction of infected cells that completely failed to express a given gene.
We limited this analysis to examining the presence / absence of the non-RNP genes in cells expressing all four RNP genes, since we might fail to detect viral transcripts that are actually present at low levels in RNP-deficient cells due to the lower viral burden in these cells.
At the 8- and 10-hour time points, between 5\% and 17\% of cells fail to express any one of the four non-RNP genes (\FIG{fluburdenbyflugene}C, \FIG{fluburdenbyflugene}-source~data~\ref{figdata:fracmissing}).
The absence of a given gene appears to be an independent event, as the probability of observing all four non-RNP genes in a cell is well predicted by simply multiplying the probabilities of observing each gene individually (\FIG{fluburdenbyflugene}C and \FIG{fluburdenbyflugene}-source~data~\ref{figdata:fracmissing}). 
If we extrapolate the frequencies at which cells lack non-RNP genes to the RNP genes, then we would predict that 35-50\% of infected cells express mRNAs from all eight genes.
This estimate of the frequency at which infected cells express mRNAs from all eight gene segments is slightly higher than previous estimates of 13\%~\citep{Brooke:2013kb} and 20\%~\citep{Dou:2017cp}.
At least one difference is that \citet{Brooke:2013kb} stained for proteins whereas we examined the expression of mRNAs -- it is likely that some cells contain mutated viral genes that fail to produce stable protein even when mRNA is expressed.
	
\subsection{The relative amounts of different viral mRNAs are more consistent across cells.}
The results above show that the amount of viral mRNA in infected cells varies over several orders of magnitude.
Does the relative expression of viral genes exhibit similar cell-to-cell variability?
To address this question, we focused on cells that derived $>$5\% of their mRNA from virus, since estimates of relative viral gene expression will be less noisy in cells with more viral mRNAs.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_flu_expr_manualedits.pdf}}
\caption{\label{fig:fluexpr}
Relative expression of influenza virus genes in highly infected cells (>5\% of total mRNA from virus).
{\bf (A)} 
The fraction of viral mRNA from each viral gene for each cell. 
{\bf (B)}
Box plots showing the distribution of the fraction of viral mRNA per cell from each viral gene.
The black lines at the notches are the medians, and the tops and bottoms of boxes indicate the first and third quartiles.
Whiskers extend to the highest or lowest data point observed within 1.5x the interquartile range, outliers shown as circles.
Notches extend 1.58x the interquartile range divided by the square root of the number of observations. 
{\bf (C)}, {\bf (D)} 
The same plots, but only including cells for which we observed at least one molecule of each viral gene.
}
\figdata{The raw data for all cells are in \texttt{p\_flu\_expr\_all.csv}.}
\figdata{The raw data for fully infected cells are in \texttt{p\_flu\_expr\_fullyinfected.csv}.}
\end{figure}

In contrast to the extreme variability in the total viral mRNA per cell, the fraction of this viral mRNA derived from each gene is much more consistent across cells (\FIG{fluexpr}A).
Total viral mRNA varies by orders of magnitude, but the fraction from any given viral gene is fairly tightly clustered around the median value for all cells (\FIG{fluexpr}B).
The relative levels of each viral mRNA in our cells are similar to prior bulk measurements made by Northern blots~\citep{hatada1989control}, which also found an expression hierarchy of M $>$ NS $\gg$ NP $>$ NA $>$ HA $\gg$ PB2 $\sim$ PB1 $\sim$ PA.
The cell-to-cell consistency in the relative expression of different viral genes is even tighter if we limit the analysis only to cells that express all eight viral genes (\FIG{fluexpr}C,D).
Therefore, with the exception of complete gene absence, the factors that drive the dramatic cell-to-cell variability in the amount of viral mRNA have roughly similar effects on all viral genes in a given cell.
This finding is consistent with prior work showing positive correlations among the abundance of several viral genome segments in individual cells~\citep{Heldt:2015iz}.

\subsection{Co-infection can provide infected cells with the full complement of viral genes.}
Our sequencing enables us to identify the rare cells that were co-infected with both wild-type and synonymously barcoded viral variants.
Overall, we captured 10 such co-infected cells that had $>$5\% of their mRNA derived from virus (\FIG{coexpression}).
Seven of these 10 cells expressed all eight viral genes.
The majority (4 of 7) of these cells would \emph{not} have expressed all the viral genes in the absence of co-infection, since they have at least one gene exclusively derived from each viral variant.
For instance, the cell with 11.2\% of its mRNA from virus in the upper right of \FIG{coexpression} expresses M only from the wildtype viral variant, and NP and HA only from the synonymously barcoded variant.
Our data therefore provide the first direct single-cell observation of the fact that co-infection can rescue missing viral genes~\citep{Brooke:2013kb,brooke2014influenza,Fonville:2015cg,aguilera2017plaques}.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_coinfection.pdf}}
\caption{\label{fig:coexpression}
The abundance of each viral transcript in cells that are co-infected with the two viral variants and have $>$5\% of their mRNA derived from virus.
The bars show the logarithms of the numbers of each viral mRNA detected, and are colored in linear proportion to the fraction of that mRNAs derived from wild-type or synonymously barcoded virus.
}
\figsupp[Co-infected cells express roughly equal amounts of a gene from each infecting viral variant. \label{figsupp:flowcyto}]{
{\bf(A)} Cells were co-infected with a mix of wild-type virus and virus in which the HA gene was replaced by GFP flanked by the terminal regions of the HA gene segment. 
At 10 hours post-infection, cells were analyzed by flow cytometry for HA and eGFP expression.
{\bf(B)} The expression of HA and GFP are correlated in co-infected cells. 
Shown are the quantile-normalized HA and eGFP signals for double-positive cells. 
Cells are colored by density, using a Gaussian kernel density estimate. 
{\bf(C)},{\bf(D)},{\bf(E)} Gating controls, single infection with eGFP virus, single infection with wild-type virus, and uninfected cells, respectively. 
}{\includegraphics[width=\linewidth]{figures/Pseudovirus_flow_cytometry/coinfFlow_D02.png}}
\figdata{The raw data plotted in this figure are in \texttt{p\_co-infection.csv}.}
\figdata{The sequence of the HA viral RNA carrying the GFP gene is in \texttt{HAflank-eGFP.fasta}.\label{figdata:HAflank}}
\end{figure} 

Another observation from \FIG{coexpression} is that co-infected cells usually express roughly equal amounts of transcripts from each of the two viral variants.
This observation is consistent with the finding by \citet{Dou:2017cp} and \citet{Huang:2008gy} that the temporal window for co-infection is short -- if both viral variants infect a cell at about the same time, then neither will have a headstart and so each will have a roughly equal opportunity to transcribe its genes.

To support this idea with a larger dataset albeit at lower resolution, we generated a virus in which the HA coding sequence was replaced by GFP.
We then co-infected cells with a mix of wildtype and $\Delta$HA-GFP virus and used flow cytometry to score cells for the presence of HA only (infection by wildtype virus), GFP only (infection by $\Delta$HA-GFP virus), or both (co-infection) as shown in \FIGSUPP[coexpression]{flowcyto}.
As in our single-cell sequencing data, we found that expression of HA and GFP were highly correlated, indicating that co-infected cells typically expressed roughly equal amounts of transcript from each viral variant.

\subsection{Activation of the interferon response is rare in single infected cells.}
Because our sequencing captured all polyadenylated transcripts, we can examine whether there are prominent changes in the host-cell transcriptome in sub-populations of infected cells.
Influenza virus infection can trigger innate-immune sensors that lead to the transcriptional induction of type I and III interferons, and subsequently of anti-viral interferon-stimulated genes~\citep{Killip:2015dw,Iwasaki:2014dw,crotta2013type}.
However, activation of the interferon response is stochastic and bi-modal at the level of single cells~\citep{Chen:2010cr,shalek2013single,shalek2014single,PerezCidoncha:2014jr,bhushal2017cell,hagai2017gene}.
We therefore hypothesized that we might see two sub-populations of infected cells: one in which the interferon response inhibited viral transcription, and another in which the virus was able to express high levels of its mRNA by evading or blocking this response.

\begin{figure}
\centerline{\includegraphics[width=0.9\linewidth]{figures/p_small_tsne_merge_manualedits.pdf}}
\caption{\label{fig:tsne}
A t-SNE plot created by semi-supervised clustering using genes that co-vary with viral infection status.
Each point is a single cell, and each panel shows an identical layout but colors the cells according to a different property.
{\bf (A), (B)}
Cells colored by the fraction of their mRNA derived from virus.
{\bf (C)}
Cells colored by the experimental sample.
{\bf (D)}
Cells colored by the number of detected transcripts from type I and III interferons (IFN).
Only one cell has detectable interferon expression (in orange, indicated with arrow).
{\bf (E)}
Cells colored by the expression of the interferon-stimulated gene IFIT1.
{\bf (F)}
Cells colored by whether they express the viral NS gene.
The one interferon-positive cell is lacking NS, but so are many interferon-negative cells.
}
\end{figure}

To examine whether there were distinct sub-populations of virus-infected cells, we used a semi-supervised t-SNE approach~\citep{VanderMaaten:2008tm} to cluster cells by genes that co-varied with viral infection status.
As shown in \FIG{tsne}A,B, this approach effectively grouped cells by the amount of viral mRNA that they expressed.
Sample-to-sample variation was regressed away during the clustering, as cells did not obviously group by time-point, with expected exception that the uninfected and 6-hour samples had few cells in the region of the plot corresponding to large amounts of viral mRNA (\FIG{tsne}C).

But to our surprise, we did not see a prominent clustering of infected cells into sub-populations as expected if the interferon response was strongly activated in some cells.
To investigate further, we annotated each cell by the total number of type I and III interferon transcripts detected.
Remarkably, only a single cell expressed detectable interferon (\FIG{tsne}D).
We also examined interferon-stimulated genes, which are induced by autocrine and paracrine interferon signaling.
\FIG{tsne}E shows expression of one such gene, IFIT1~\citep{Fensterl:2011fp}.
As with interferon itself, expression of IFIT1 was rare and most prominent in the single interferon-positive cell, presumably due to the higher efficiency of autocrine versus paracrine signaling.
Notably, interferon and interferon-stimulated genes were also relatively ineffective at blocking viral transcription in the single cell in which they were potently induced, since $>$10\% of the mRNA in this cell was derived from virus (\FIG{tsne}A,B,D,E).

We posited that the paucity of interferon induction might be due to the activity of influenza virus's major interferon antagonist, the NS1 protein~\citep{garcia1998influenza,hale2008multifunctional}.
We therefore identified cells that expressed substantial amounts of viral mRNA but lacked the NS gene (\FIG{tsne}F).
Consistent with the idea that NS1 is important for suppressing interferon, the one interferon-positive cell lacked detectable expression of the NS gene.
But other cells that lacked NS expression still failed to induce a detectable interferon response, despite often having a substantial amount of their mRNA derived from virus (\FIG{tsne}).
This result is in line with other work showing that NS1-deficient influenza virus does not deterministically induce interferon~\citep{Killip:2017ef,Kallfass:2013kp}.
Therefore, many individual infected cells fail to activate innate-immune responses even when the virus lacks its major interferon antagonist.

\subsection{Some host genes co-vary with viral gene expression.}
We examined whether any host genes were differentially expressed in cells with more viral mRNA.
We restricted this analysis to infected cells with all eight viral genes in order to focus on cellular genes that were associated with viral mRNA burden independent of effects due to the presence or absence of particular viral transcripts.
We identified 43 cellular genes that co-varied with viral mRNA expression at a false discovery rate of 0.1 (\FIG{cellulargenes}, \FIG{cellulargenes}-source~data~\ref{figdata:cellulargenes}).

\begin{figure}[t!]
\centerline{\includegraphics[width=0.65\linewidth]{figures/p_cellular_heatmap_manualedits.pdf}}
\caption{\label{fig:cellulargenes}
Cellular genes that co-vary in expression with the amount of viral mRNA in cells expressing all eight viral genes.
The columns are cells, ordered from left to right by the fraction of mRNA derived from virus.
Each row is a gene that is differentially expressed as a function of the fraction of mRNA derived from virus at a false discovery rate of 0.1. 
Genes for which the color goes from blue at left to red at right are expressed at higher levels in cells with more viral mRNA.
The scale bar indicates the number of standard deviations above or below the mean expression, truncated at 3-fold on both sides. 
}

\figdata{The full results of the differential expression test are in \texttt{p\_sig\_cellular\_genes.csv}. \label{figdata:cellulargenes}}

\figdata{A gene-set analysis for pathways associated with the amount of viral mRNA is in \texttt{p\_pathway\_enrichment.csv}. \label{figdata:geneset}}

\figsupp[Many genes that co-vary with viral load are involved in the oxidative stress response. \label{figsupp:ROStable}]{
Table delineating genes in \FIG{cellulargenes} that are associated with the response to oxidative stress~\citep{Duong:2017ju,Jung:2017jz,Lee:2017ba,Peuchant:2017hj,MacLeod:2016kt,Jiang:2016cf,Gorrini:2013jv,Miura:2013fr,Kim:2009gi,Banning:2005cz,Murray:2003ie,Doyle:1999ts}.
}{\includegraphics[width=0.7\linewidth]{figures/ROStable.pdf}}

\figsupp[Pre-treating to reduce oxidative stress decreases the fraction of infected cells expressing detectable viral protein. \label{figsupp:oxidative}]
{Cells were treated as indicated with 15 mM N-acetylcysteine (NAC), an antioxidant, and infected at an MOI of 0.1 as calculated by TCID50 on MDCK-SIAT1 cells. 
At 10 hours post-infection, cells were analyzed by flow cytometry for HA expression.
The percentage of HA-positive cells is indicated on the flow cytometry plots.
Data are shown for three independent replicates.
}{\includegraphics[width=0.7\linewidth]{figures/antiox.jpg}}

\end{figure}

A gene-set analysis shows that many cellular genes that are associated with the amount of viral mRNA are involved in the response to reactive oxygen species or hypoxia (\FIG{cellulargenes}-source~data~\ref{figdata:geneset}). 
Genes known or suspected to be regulated by the Nrf2 master regulator in response to oxidative stress are often expressed at higher levels in cells with more viral mRNA (\FIG{cellulargenes}).
These genes produce proteins that are involved in detoxification of reactive oxygen species or resultant products, the management of misfolded proteins, the electron transport chain, or a general stress response (\FIGSUPP[cellulargenes]{ROStable}). 
We additionally see reduced expression of the nitric oxide synthase interacting protein (NOSIP). 
Transient oxidative stress is known to occur during viral infection, and may act in a proviral fashion via MAPK activation driving vRNP export~\citep{Amatore:2014cs}.
The antioxidant response is thought to be largely antiviral, potentially through inhibition of MAPK activity \citep{Lin:2016ec,Sgarbanti:2014ht}.
To directly test the effect of transient oxidative stress, we compared the fraction of cells that expressed detectable viral protein when infected either with or without pre-treatment to suppress oxidative stress.
\FIGSUPP[cellulargenes]{oxidative} shows that the cells pre-treated with an antioxidant exhibited less frequent detectable expression of viral protein.
These results, in conjunction with the differential expression test in \FIG{cellulargenes} and the prior work mentioned above, suggest that oxidative stress acts in a proviral fashion. 

The gene-set analysis also found that the amount of viral mRNA was associated with the expression of genes involved in the G2-M cell-cycle checkpoint (\FIG{cellulargenes}-source~data~\ref{figdata:geneset}).
The cell-cycle associated genes CCND3, MKI67, UBE2S, and CENPF are all expressed at significantly lower levels in cells with more viral mRNA (\FIG{cellulargenes}).
However, our data are not sufficient to determine whether the lower expression of these genes is a cause or effect of the reduction in viral mRNA.

Interestingly, none of the cellular genes that are significantly associated with the amount of viral mRNA in our study are among the 128 genes that \citet{watanabe2010cellular} report as having been identified multiple times in genome-wide screens for factors affecting influenza virus replication.
One possible explanation is that most of the cell-to-cell heterogeneity in our experiments might arise from viral segment absence or mutations, pure stochasticity, or more subtle alterations in host-cell state -- not due to changes in expression of the type of single large-effect genes that are usually identified in genome-wide knockdown / knockout studies.

\section{Discussion}
We have quantified the total transcriptome composition of single cells infected with influenza virus.
While we observe a general increase in the amount of viral mRNA over time as expected from bulk measurements~\citep{hatada1989control,Shapiro:1987ur}, there is wide variation in viral gene expression among individual infected cells.

The most obvious form of heterogeneity is the complete failure of some infected cells to express one or more viral genes, which we estimate occurs in about half the infected cells in our experiments.
The absence of some viral genes in some infected cells has been noted previously~\citep{Brooke:2013kb,Heldt:2015iz,Dou:2017cp}, and our work provides a holistic view by quantifying the total viral transcriptional load as a function of the level of each mRNA.
We find that cells lacking expression of any of the four genes that encode the viral RNP express much less total viral mRNA, consistent with prior bulk studies~\citep{Vreede:2004ip,eisfeld2015centre}.
Interestingly, the \emph{reason} some cells fail to express some viral genes remains unclear.
The prototypical influenza virion packages one copy of each of the eight gene segments~\citep{noda2006architecture,hutchinson2010genome}, but some virions surely package fewer~\citep{brooke2014influenza}.
However, it is also possible that much of the viral gene absence is due to stochastic loss of viral RNPs after infection but prior to the initiation of viral transcription in the nucleus.

The absence of viral genes only partially explains the cell-to-cell variation in amount of viral mRNA, which still varies from $<$1\% to $>$50\% among cells expressing all the viral genes.
It is likely that other viral genetic factors explain some of this remaining heterogeneity.
The 3'-end sequencing strategy used in our experiments detects the presence of a viral gene, but does not identify whether that gene contains a mutation that might hinder viral replication.
However, viral mutations are also unlikely to explain all the observed heterogeneity, since current consensus estimates of influenza virus's mutation rate suggest that the typical virion in a stock such as the one used in our experiment should contain less than one mutation per genome~\citep{parvin1986measurement,suarez1992heterogeneity,suarez1994estimation,nobusawa2006comparison,bloom2014experimentally,pauly2017novel}.

The rest of the heterogeneity must be due to some combination of cellular factors and inherent stochasticity.
Some features of the cellular transcriptome co-vary with the amount of influenza mRNA. 
In particular, the viral load in individual cells is associated with the expression of genes involved in response to cellular stresses, including oxidative stress.
It will be interesting to determine if these cellular transcriptional signatures are simply a consequence of the stress imposed by viral replication, or if their stronger activation in some cells is a causative factor that promotes viral transcription.
However, it also would not be surprising if a substantial amount of the cell-to-cell heterogeneity cannot be ascribed to pre-existing features of either the viral genome or cellular state. 
Apparently stochastic heterogeneity is a common feature of many processes at a single-cell level~\citep{cai2006stochastic,raj2006stochastic,buganim2012single,shalek2013single,avraham2015pathogen} -- especially when those processes are initiated by very small numbers of initial molecules~\citep{elowitz2002stochastic}, as is the case for low-MOI viral infection.

Our data do suggest that the factors driving the heterogeneity in viral transcriptional load exert relatively concordant effects on all viral genes in a given cell.
Specifically, despite the extreme heterogeneity in total viral mRNA per cell, the relative levels of the viral mRNAs are reasonably consistent across cells, and generally reflective of classical bulk measurements~\citep{hatada1989control}.
Therefore, despite the stochasticity inherent in initiating transcription and replication of each gene from a single copy carried by the incoming virion, as long as a gene is not completely lost then the virus possesses mechanisms to control its relative expression~\citep{Shapiro:1987ur,hatada1989control,perez2010influenza,heldt2012modeling,chua2013influenza}.

One factor that surprisingly does \emph{not} appreciably contribute to the heterogeneity in our experiments is activation of innate-immune interferon pathways.
Only one of the hundreds of virus-infected cells expresses any detectable interferon, despite the fact that a number of cells fail to express the influenza-virus interferon antagonist NS1.
It is known that interferon activation is stochastic at the level of single cells in response to both synthetic ligands~\citep{shalek2013single,shalek2014single,bhushal2017cell,hagai2017gene} and actual infection~\citep{Rand:2012kl,PerezCidoncha:2014jr,avraham2015pathogen,Killip:2017ef}.
But interferon expression is a prominent transcriptional signature of high-MOI influenza virus infection of bulk cells, including in the epithelial cell line and at the time-points used in our experiments~\citep{Geiss:2002a,sutejo2012activation}.
So it is notable how rarely single cells express interferon.
Interferon expression would surely be more common at later times or with a viral stock passaged at higher MOI, since paracrine interferon signaling~\citep{crotta2013type} and accumulation of defective viral particles enhance innate-immune detection~\citep{Tapia:2013kf,Lopez:2014en}.
However, the early events of physiological influenza infection involve just a few virions~\citep{varble2014influenza,mccrone2017evolutionary}, and so it is interesting to speculate whether rare events such as interferon activation during the first few cycles of viral replication could contribute to heterogeneity in the eventual outcome of infection.

Overall, our work shows the power and importance of characterizing cellular infection at the level of single cells~\citep{avraham2015pathogen}.
Viral infection can involve heterogeneity in the genetic composition of the incoming virion, the host-cell state, the bi-modality of innate-immune activation, and the inherent stochasticity of molecular processes initiated by a single copy of each viral gene.
In our experiments with short-timeframe and low-MOI infections with a relatively pure stock of influenza virus, we find only a minor role for innate-immune activation, but a substantial role for heterogeneity in the complement of viral genes that are expressed in individual cells and at least some contribution of host-cell state.
Our current experiments are not able to quantify the role of other possibly important factors such as mutations in viral genes, but we suspect that they may also contribute.
Future extensions of the approaches described here should enable further deconstruction of the sources of cell-to-cell heterogeneity during viral infection, thereby enabling a deeper understanding of how the bulk features of infection emerge from processes within individual virus-infected cells.

\section{Methods and Materials}

\subsection{Cell lines and viruses}
The following cell lines were used in this study: the human lung epithelial carcinoma line A549 (ATCC CCL-185), the MDCK-SIAT1 variant of the Madin Darby canine kidney cell line overexpressing human SIAT1 (Sigma-Aldrich 05071502), and the human embryonic kidney cell line 293T (ATCC CRL-3216). 
The A549 cells were tested as negative for mycoplasma contamination by the Fred Hutch Genomics Core, and authenticated using the ATCC STR profiling service.
All cells were maintained in D10 media (DMEM supplemented with 10\% heat-inactivated fetal bovine serum, 2 mM L-glutamine, 100 U of penicillin/ml, and 100 \si{\micro}g of streptomycin/ml) at 37 \si{\degreeCelsius} at 5 \% CO\textsubscript{2}.

Wildtype A/WSN/1933 (H1N1) influenza virus was generated by reverse genetics using the plasmids pHW181-PB2, pHW182-PB1, pHW183-PA, pHW184-HA, pHW185-NP, pHW186-NA, pHW187-M, and pHW188-NS~\citep{hoffmann2000dna}.
The sequences of the viral RNAs encoded in these plasmids are in \FIG{workflow}-source~data~\ref{figdata:viralsequences}.
Reverse-genetics plasmids encoding the synonymously barcoded WSN virus were created by using site-directed mutagenesis to introduce two synonymous mutations near the 3' end of the mRNA for each viral gene.
The sequences of the synonymously barcoded viral RNAs are in \FIG{workflow}-source~data~\ref{figdata:viralsequences}.

To generate viruses from these plasmids, we transfected an equimolar mix of all eight plasmids into cocultures of 293T and MDCK-SIAT1 cells seeded at a ratio of 8:1. 
At 24 hours post-transfection, we changed media from D10 to influenza growth media (Opti-MEM supplemented with 0.01\% heat-inactivated FBS, 0.3\% BSA, 100 U of penicillin/ml, 100  \si{\micro}g of streptomycin/ml, and 100 \si{\micro}g of calcium chloride/ml).
At 48 hours post-transfection we harvested the virus-containing supernatant, pelletted cellular material by centrifugation at 300 x g's for 4 minutes, and stored aliquots of the clarified viral supernatant at -80 \si{\degreeCelsius }. 
We then titered thawed aliquots of viral by TCID50 on MDCK-SIAT1 cells, computing titers via the formula of \citet{reed1938simple}.
To generate our ``high-purity'' stocks of viruses for the single-cell sequencing experiments, we then infected MDCK-SIAT1 cells at an MOI of 0.01, and let the virus grow for 36 hours prior to harvesting aliquots that were again clarified by low-speed centrifugation, aliquoted, stored at -80  \si{\degreeCelsius }, and titered by TCID50.
The high-MOI passage (high-defective particle) stock used in \FIG{viruspopulations} was generated by instead passaging in MDCK-SIAT1 cells twice at an MOI of 1 for 48 hours.

For the experiments in \FIGSUPP[coexpression]{flowcyto}, we created a virus that carried an HA gene segment in which GFP replaced most of the HA coding sequence, following a scheme first described by \citet{marsh2007specific}.
Briefly, we created a plasmid encoding a viral RNA with GFP in place of the HA coding sequence in the context of the pHH21~\citep{Neumann:1999ws} reverse-genetics plasmid, removing potential start codons upstream of the GFP (see \FIG{coexpression}-source~data~\ref{figdata:HAflank} for the sequence of the viral RNA).
We then generated GFP-carrying virus by reverse-genetics in cells constitutively expressing HA~\citep{Doud:2016gm}.
To obtain sufficient titers, this HA-eGFP virus was expanded for 44 rather than 36 hours after initiating infection at an MOI of 0.01.

\subsection{qPCR}
For the qPCR in \FIG{viruspopulations} and \FIGSUPP[fluburdenbyflugene]{cyclohexamide}, A549 cells were seeded at 3$\times$10\textsuperscript{5} cells per well in a 6-well tissue culture plate in D10 the day prior to infection. 
On the day of infection, a single well was trypsinized and the cells were counted in order to determine the appropriate amount of virus to use to achieve the intended MOI.
Immediately before infection, D10 was replaced with influenza growth media.
For cells incubated with cyclohexamide, the compound was added to a final concentration of 50  \si{\micro}g/ml at the time of infection -- previously confirmed to be sufficient to halt viral protein production~\citep{Killip:2014kz}.
RNA was purified using the QIAGEN RNeasy plus mini kit following manufacturer's instructions. 
cDNA was synthesized using an oligoDT primer and the SuperScript\textsuperscript{TM} III first-strand synthesis supermix from ThermoFisher using the manufacturer's protocol. 
Transcript abundance was measured using SYBR\textsuperscript{TM} green PCR master mix, using a combined anneal/extension step of 60 \si{\degreeCelsius } for one minute with the following primers: \emph{HA}: 5'-GGCCCAACCACACATTCAAC-3', 5'-GCTCATCACTGCTAGACGGG-3', \emph{IFNB1}: 5'-AAACTCATGAGCAGTCTGCA-3', 5'-AGGAGATCTTCAGTTTCGGAGG-3', \emph{L32}: 5'-AGCTCCCAAAAATAGACGCAC-3', 5'-TTCATAGCAGTAGGCACAAAGG-3'.   
Biological triplicates were performed for all samples.

For the measurements of viral genomic HA content in \FIG{viruspopulations}A, vRNA was harvested from 80 \si{\micro}l of viral supernatant by the addition of 600 \si{\micro}l of RLT plus before proceeding with the standard QIAGEN RNeasy Plus Mini kit protocol.
The cDNA was generated using SuperScript\textsuperscript{TM} III first-strand synthesis supermix using the manufacturer's protocol, and using the universal vRNA primers of \citet{Hoffmann:2001vj} with the modifications described in  \citet{Xue:2017dl}.
The qPCR was then performed as for mRNA measurements.
A standard curve was generated from three independent dilutions of the HA-encoding reverse genetics plasmid. 
All vRNA values represent three independent RNA extractions with two replicate qPCR measurements. 

\subsection{Flow cytometry titering and analyses}
To determine viral titers in terms of HA-expressing units and for the flow cytometry, A549 cells were seeded in a 6-well plate and infected as described above for the qPCR analyses.
Cells were harvested by trypsinization, resuspended in phosphate-buffered saline supplemented with 2\% heat-inactivated FBS, and stained with 10 \si{\micro}g/ml of H17-L19, a mouse monoclonal antibody confirmed to bind to WSN HA in a prior study~\citep{Doud:2017bw}.
After washing in PBS supplemented with 2\% FBS, the cells were stained with a goat anti-mouse IgG antibody conjugated to APC.
Cells were then washed, fixed in 1\% formaldehyde, and washed further before a final resuspension and analysis. 
We then determined the fraction of cells that were HA positive and calculated the HA-expressing units.
For NS1 staining, cells stained for HA as described above were permeabilized using BD Cytofix/Cytoperm following manufacturer's instructions, stained with anti-NS1 (GTX125990, Genetex) at 4.4 \si{\micro}g/ml, washed,  stained with a  goat anti-rabbit IgG antibody conjugated to Alexa Fluor 405, washed, and analyzed.
To analyze the effect of N-acetylcysteine, the compound was added to cells in D10 9h prior to media change and infection, and included in infection media.
Stocks of N-acetylcysteine were reconstituted immediately prior to use, and the pH of growth media supplemented with this compound was adjusted using sodium hydroxide.
After channels were compensated and cells gated to exclude multiplets and debris in FlowJo, data were extracted using the R package flowCore~\citep{LeMeur:2007uo} and analyzed using a custom Python script.
Guassian kernel density estimates were obtained using the scipy stats package method, guassian\_kde, using automatic bandwidth determination~\citep{vanderWalt:2017dp}.
For gating on NS1 positive cells, the percentage of influenza-infected cells was determined by HA staining alone, and the top quantile of NS1-stained cells matching that percentage were taken as the NS1 positive population.


\subsection{Infections for single-cell mRNA sequencing}
Single-cell sequencing libraries were generated using the 10x Chromium Single Cell 3' platform~\citep{zheng2017massively} using the V1 reagents.

All time points except for the second 8-hour sample (8hr-2) were prepared on the same day.
For the infections, A549 cells were seeded in a 6-well plate, with two wells per time point. 
A single well of cells was trypsinized and counted prior to initiation of the experiment for the purposes of calculating MOI.
Wild-type and synonymously barcoded virus were mixed to an estimated ratio of 1:1 based on prior, exploratory, single-cell analyses (data not shown). 
At the initiation of our experiment, the wells for all time points were changed from D10 to influenza growth media.
Cells were then infected with 0.3 HA-expressing units of virus per cell (a determined by flow cytometry).
The infections were performed in order of time point: first the 10-hour time point, then the 8-hour, and then the 6-hour time point.
At one hour after infection, the media for each time point was changed to fresh influenza growth media.
Note that the HA-expressing units were calculated without this additional washing step, and so likely represent an overestimate of our final infectious dose (consistent with the fact that fewer than 30\% of cells appear infected in the single-cell sequencing data).
All cells were then harvested for single-cell analysis concurrently -- ensuring all had spent equivalent time in changed media .
For 8hr-2 sample, cells were infected as above except that the cells were infected at 0.1 HA-expressing units of virus per cell but no wash step was performed, and the sample was prepared on a different day.
After harvest, cells were counted using disposable hemocytometers and diluted to equivalent concentrations with an intended capture of 3000 cells/sample following the manufacturer's provided by 10x Genomics for the Chromium Single Cell platform.
All subsequent steps through library preparation followed the manufacturer's protocol.
Samples were sequenced on an Illumina HiSeq. 

\subsection{Computational analysis of single-cell mRNA sequencing data}
Jupyter notebooks that perform all of the computational analyses are available in Supplementary~file~\ref{suppfile:code} and at \url{https://github.com/jbloomlab/flu_single_cell}~\citep[][copy archived at \url{https://github.com/elifesciences-publications/flu_single_cell}]{russell2018github}. 

Briefly, the raw deep sequencing data were processed using the 10X Genomics software package CellRanger (version 2.0.0).
The reads were aligned to a concatenation of the human and influenza virus transcriptomes.
The human transcriptome was generated by filtering genome assembly GRCh38 for protein coding genes defined in the GTF file GRCh38.87.
The influenza virus transcriptome was generated from the reverse-complement of the wildtype WSN viral RNA sequences as encoded in the reverse-genetics plasmids (\FIG{workflow}-source~data~\ref{figdata:viralsequences}).
The aligned deep sequencing data are available on the GEO repository under accession GSE108041 (\url{https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE108041}).

CellRanger calls cells based on the number of observed cell barcodes, and creates a cell-gene matrix.
We used custom Python code to annotate the cells in this matrix by the number of viral reads that could be assigned to the wildtype and synonymously barcoded virus.
Only about half of the viral reads overlapped the barcoded regions of the genes (\FIG{workflow}A) and could therefore be assigned to a viral barcode (\FIGSUPP[viralbarcodes]{barcodescalled}).
So for calculations of the number of reads in a cell derived from each viral barcode for each viral gene, the total number of detected molecules of that gene are multiplied by the fraction of those molecules with assignable barcodes that are assigned to that barcode.
This annotated cell-gene matrix is in Supplementary~file~\ref{suppfile:cellgenematrix}.
A Jupyter notebook that performs these analyses is in Supplementary~file~\ref{suppfile:code}.

The annotated cell-gene matrix was analyzed in R, primarily using the Monocle package (version 2.4.0)~\citep{Qiu:2017dw,Trapnell:2014kg}.
A Jupyter notebook that performs these analyses is in Supplementary~file~\ref{suppfile:code}.
For each sample, cell barcodes that had $>$2.5-fold fewer or more UMI counts mapping to cellular transcripts than the sample mean were excluded from downstream analyses (see red vertical lines in \FIG{cells}B).

In order to determine an appropriate cutoff for how many reads a cell needed to contain in order to be classified as infected, we calculated the mean viral barcode purity across all cells that contained at least a given fraction of viral mRNA and had multiple viral reads that could be assigned a barcode (\FIG{viralbarcodes}B,C and \FIGSUPP[viralbarcodes]{thresholds}).
We then determined the threshold fraction of viral mRNA at which the mean purity no longer increased as a function of the amount of viral mRNA.
This threshold represents the point at which we have effectively eliminated cells that have low barcode purity simply due to lysis-acquired reads sampled randomly from both viral barcodes.
As is apparent from \FIG{viralbarcodes}B, only the 10-hour sample and the 8hr-2 sample have the excess of mixed barcodes among cells with low amounts of viral mRNA.
The likely reason is that these samples have more total viral mRNA (and so there is more available mRNA to be acquired from lysed cells); in addition, there is always some experimental variability in the amount of cell lysis during the 10X sequencing process, and these samples may simply have the most.
So the above threshold procedure is appropriate for those two samples.
For the other samples, we simply set a minimum threshold of requiring at least a fraction 2$\times$$10^{-4}$ reads to come from viral mRNA as explained in the legend to \FIGSUPP[viralbarcodes]{thresholds}.
The thresholds for each sample are shown in \FIG{viralbarcodes}C and \FIGSUPP[viralbarcodes]{thresholds}.
This procedure is expected to be conservative, and may miss some truly infected cells with very low amounts of viral mRNA.
For subsequent analyses, we retained all infected cells and a subsample of uninfected cells (the greater of 50 or the number of infected cells for that sample).
The rationale for subsampling the uninfected cell is that the vast majority of cells are uninfected, and we did not want these cells to completely dominate the downstream analyses.
Cells were classified as co-infected if both viral variants had an RNA level that exceeded the threshold, and if the minor variant contributed at least 5\% of the viral mRNA.

For the semi-supervised t-SNE clustering, we used Monocle's cell hierarchy function to bin cells into those with no viral mRNA, <2\% viral mRNA, between 2\% and 20\% viral mRNA, and >20\%. 
Candidate marker genes for t-SNE dimensionality reduction were then determined using the Monocle function markerDiffTable, excluding the effects of sample variation and the number of genes identified in a given cell, using a q-value cutoff of 0.01.
The specificity of these markers was determined using the function calculateMarkerSpecificity -- the top 50 markers were retained, and used to place populations in a two-dimensional plane based on tSNE dimensionality reduction.

For the analyses of cellular genes that differed in expression as a function of the amount of viral mRNA, we only considered cells that expressed all 8 viral mRNAs to avoid effects driven simply by viral gene absence.
We also only considered cellular genes in the differential gene analysis, since viral gene expression will tautologically co-vary with the amount of viral mRNA.
Additionally, because influenza virus has the capacity to degrade or prevent the synthesis of host mRNAs~\citep{BercovichKinori:2016iw} and contributes significantly to the total number UMIs in some cells, we calculate size factors (a scalar value representing efficiency of UMI capture) based on cellular transcripts alone. 
Finally, we assigned all cells a ceiling fraction of mRNA from virus of 25\% so that a few extremely high-expressing cells did not dominate. 
Cellular genes with expression that co-varied with the fraction of viral mRNAs in a cell were then determined using the Monocle differentialGeneTest, after removing variance explained by sample to sample variation. 
\FIG{cellulargenes} shows all genes that were significantly associated with the fraction of mRNA from virus at a false discovery rate of 0.1.
We performed the gene set analysis using the P -alues from the Monocle differentialGeneTest with \texttt{piano}~\citep{varemo2013enriching} using the hallmark gene set from GSEA v6~\citep{subramanian2005gene} and Fisher's method.

\section{Acknowledgments}
We thank Xiaojie Qiu for advice about use of the Monocle software package, David Bacsik and Robert Bradley for comments on the manuscript, and the Fred Hutch Genomics Core for performing the Illumina deep sequencing.

\bibliography{references}

\clearpage

\begin{suppfile}
\caption{\label{suppfile:code}
Computer code for the analyses.
This ZIP file contains a Jupyter notebook that runs CellRanger to align and annotate the reads, and a Jupyter notebook that uses Monocle to analyze the cell-gene matrix.
The ZIP file also includes associated custom scripts.
To just run the Monocle analysis in \texttt{monocle\_analysis.ipynb} on a pre-generated cell-gene matrix, unpack Supplementary~file~\ref{suppfile:cellgenematrix} into \texttt{./results/cellgenecounts/}.}
\end{suppfile}

\begin{suppfile}
\caption{\label{suppfile:cellgenematrix}
The annotated cell-gene matrix in Matrix Market Format.
This is the matrix generated in \texttt{./results/cellgenecounts/} by running the CellRanger analysis in \texttt{align\_and\_annotate.ipynb} in Supplementary~file~\ref{suppfile:code}.
This file is available on DataDryad at \url{https://doi.org/10.5061/dryad.qp0t3}.
}
\end{suppfile}

\end{document}

